src/components/App/App.js

```diff
@@ -20,121 +32,274 @@ function App() {
+      <CurrentUserContext.Provider value={currentUser}>
+        <Header
           isMainPage={isMainPage}
           isActive={isActive}
           setIsActive={setIsActive}
+          isHidden={isHidden}
         >
-          <SignButton
+          <Navigation
             loggedIn={loggedIn}
             isMainPage={isMainPage}
-            handleSignClick={handleSignClick}
+            isActive={isActive}
             setIsActive={setIsActive}
+          >
+            <SignButton
+              loggedIn={loggedIn}
+              isMainPage={isMainPage}
+              handleSignClick={handleSignClick}
+              setIsActive={setIsActive}
+            />
+          </Navigation>
+        </Header>
+        <Routes>
+          <Route
+            index
+            path='/'
+            element={
+              <Main
+                loggedIn={loggedIn}
+                isMainPage={true}
+                onSearch={getData}
+                articles={articles}
+                searchResult={searchResult}
+                isOpen={isCardListOpen}
+                searchErrorMessage={searchErrorMessage}
+                handleCount={handleCount}
+                showMoreButton={showMoreButton}
+                cardsCount={cardsCount}
+                onCardSave={addArticle}
+                onCardDelete={removeArticle}
+              />
+            }
           />
-        </Navigation>
-      </Header>
-      <Routes>
-        <Route
-          index
-          path="/"
-          element={<Main loggedIn={loggedIn} isMainPage={isMainPage} />}
-        />
-        <Route
-          path="/saved-news"
-          element={<SavedNews loggedIn={loggedIn} isMainPage={isMainPage} />}
+          <Route
+            path='/saved-news'
+            element={
+              <ProtectedRoute loggedIn={loggedIn}>
+                <SavedNews onCardDelete={removeArticle} />
+              </ProtectedRoute>
+            }
+          />
+        </Routes>
+        <PopupWithForm
+          onClose={closeAllPopup}
+          isOpen={isLoginPopupOpen}
+          buttonText='Sign up'
+          title='Sign in'
+          signError={signError}
+          handleAction={handleLogin}
+          switchPopup={handleSwitchPopup}
+        >
+          <FormInput type='email' name='Email' placeholder='email' />
+          <FormInput type='password' name='Password' placeholder='password' />
+        </PopupWithForm>
+        <PopupWithForm
+          onClose={closeAllPopup}
+          isOpen={isRegisterPopupOpen}
+          buttonText='Sign in'
+          title='Sign up'
+          signError={signError}
+          handleAction={handleRegister}
+          switchPopup={handleSwitchPopup}
+        >
+          <FormInput type='email' name='Email' placeholder='email' />
+          <FormInput type='password' name='Password' placeholder='password' />
+          <FormInput type='text' name='Username' placeholder='username' />
+        </PopupWithForm>
```

Comment:
according to the task you need to have 2 different components: `Login` with the markup of the login form and `Register` with the register form

So, please, put these lines inside the mentioned components